crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -159,23 +179,67 @@ where
             .channel_manager
             .list_channels()
             .iter()
-            .any(|channel| channel.counterparty.node_id == target_node);
+            .any(|channel| channel.counterparty.node_id == trader_id);
         if is_for_jit_channel {
-            let fake_scid = FakeScid::new(intercept_scid);
-            let channel = Channel::new_jit_channel(target_node, fake_scid);
+            let channel = Channel::new(
+                user_channel_id,
+                0,
+                0,
+                trader_id,
+                liquidity_request.liquidity_option_id,
+            );
             self.storage.upsert_channel(channel).with_context(|| {
                 format!(
-                    "Failed to insert shadow JIT channel for counterparty {target_node} \
-                     with fake SCID {fake_scid}"
+                    "Failed to insert shadow JIT channel for counterparty {trader_id} \
+                             with user channel id {user_channel_id}"
                 )
             })?;
         }
 
         tracing::info!(
-            peer_id = %target_node,
+            %user_channel_id,
+            %trader_id,
             interceptable_route_hint_hop = ?route_hint_hop,
             "Registered interest to open JIT channel"
         );
+        Ok(route_hint_hop)
+    }
+
+    pub fn prepare_payment_with_route_hint(&self, target_node: PublicKey) -> Result<RouteHintHop> {
+        let channels = self.channel_manager.list_channels();
+        let channel = channels
+            .iter()
+            .find(|channel| channel.counterparty.node_id == target_node)
+            .with_context(|| format!("Couldn't find channel for {target_node}"))?;
+
+        let short_channel_id = channel.short_channel_id.with_context(|| {
+            format!(
+                "Couldn't find short channel id for channel: {}, trader_id={target_node}",
+                channel.channel_id.to_hex()
+            )
+        })?;
+
+        let ldk_config = self.ldk_config.read();
+
+        let route_hint_hop = RouteHintHop {
+            src_node_id: self.info.pubkey,
+            short_channel_id,
+            fees: RoutingFees {
+                base_msat: ldk_config.channel_config.forwarding_fee_base_msat,
+                proportional_millionths: ldk_config
+                    .channel_config
+                    .forwarding_fee_proportional_millionths,
+            },
+            cltv_expiry_delta: MIN_CLTV_EXPIRY_DELTA,
+            htlc_minimum_msat: None,
+            htlc_maximum_msat: None,
+        };
+
+        tracing::info!(
+            peer_id = %target_node,
+            route_hint_hop = ?route_hint_hop,
```

Comment:
Is there a particular reason why you want `route_hint_hop` in the logs? I assume it look terrible :)